src/utils/validateFunction.ts

```diff
@@ -0,0 +1,102 @@
+import fs from 'fs-extra';
+import path from 'path';
+import fetch from 'node-fetch';
+import { Validator, ValidatorResult, ValidationError } from 'jsonschema';
+
+type FunctionDefinition = {
+  name?: string;
+  [other: string]: unknown;
+};
+
+export type Schema = {
+  $id: string;
+};
+
+export type Config = {
+  schemaUrl: string;
+  functionSchemaPath: string;
+};
+
+const fetchFunction = async (
+  functionPath: string,
+): Promise<FunctionDefinition> => {
+  const filePath = path.join(functionPath, 'function.json');
+  try {
+    const exists = await fs.pathExists(filePath);
+    if (!exists) throw new Error(`file not found ${filePath}`);
+    return (await fs.readJSON(filePath)) as FunctionDefinition;
```

Comment:
hmmm, what if the json is

```
{ "not_name": 12345 }
```

the better type might be `object` since that is all we know. Once we complete validation then we know we have a valid `FunctionDefenition` object